connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1867,7 +1928,34 @@ class SparkConnectPlanner(val session: SparkSession) {
     output.logicalPlan
   }
 
+  def transformKeyValueGroupedAggregate(rel: proto.Aggregate): LogicalPlan = {
+    val ds = UntypedKeyValueGroupedDataset(
+      rel.getInput,
+      rel.getGroupingExpressionsList,
+      java.util.Collections.emptyList())
+
+    val namedColumns = rel.getAggregateExpressionsList.asScala.toSeq.map(expr => {
+      // Use any encoder as a placeholder to perform the TypedColumn#withInputType transformation
+      val any = ds.vEncoder
+      val newExpr = new TypedColumn(transformExpression(expr), any)
+        .withInputType(ds.vEncoder, ds.dataAttributes)
+        .expr
+      Column(newExpr).named
```

Comment:
Can we just put the named functionality in a separate util function instead of creating a Typed column here?